src/main/webapp/script.js

```diff
@@ -26,54 +42,101 @@ function loadNotifications() {
 }
 
 /**
- * Appends child to navbar dropdown. Represents a notification.
- * @param {Notification} notification
+ * Fetches questions from server, wraps each in an <li> element, 
+ * and adds them to the DOM.
  */
-function createNotificationsElement(notification) {
-  // Create a link to redirect the user to the question that was answered or commented.
-  const linkElement = document.createElement('a');
-  linkElement.innerText = linkElement.innerText.concat(notification.message, " - ");
-  linkElement.innerText = linkElement.innerText.concat(notification.timestamp.toString());
-  linkElement.setAttribute("href", notification.url);
-  // Create list element.
-  const liElement = document.createElement('li');
-  liElement.appendChild(linkElement);
-  liElement.setAttribute("class","list-group-item");
-  return liElement;
+async function fetchForum() {
+  const response = await fetch('/fetch-forum');
+  const questionsObject = await response.json();
+  const questionsContainer = document.getElementById('forum');
+  questionsObject.forEach(question => {
+    questionsContainer.appendChild(createQuestionElement(question));
+  });
 }
 
 /**
- * Creates notification when an answer or comment is posted.
- * @param {string} type
- * @param {int} id
+ * Displays navbar authentication and inbox buttons according to login status.
  */
-function notify(type, id) {
-  fetch('notification?type=' + type + '&modifiedElementId=' + id, {
-    method: 'POST'
+function fetchAuthentication() {
```

Comment:
I think we should also order them in the order they are called, so fetchAuthentication() would go before fetchForum().

Reply:
done